jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -90,6 +90,20 @@ public static RetainableByteBuffer toRetainableByteBuffer(Resource resource, Byt
             }
         }
 
+        // Optimize for ContentSourceResource.
+        if (resource instanceof ContentSourceResource contentSourceResource)
+        {
+            try
+            {
+                ByteBuffer buffer = Content.Source.asByteBuffer(contentSourceResource.newContentSource(bufferPool, 4096, direct, 0L, -1L));
```

Comment:
Why `4096` again?

Reply:
A default has to be used, and `4096` is used by the `InputStream` fallback code later in that same method. Having default buffer sizes popping up at random is a weakness of 12.0.x we're aware of that was sorted in 12.1.x, so this isn't going to be cleaned up here.

Reply:
I'm not following anymore...

If we have buffers in the buffer pool that are sufficiently large, we should simply use them in full.

There should not be fallback code using a hardcoded `4096` at all, we should always get the buffer config set for the server, e.g.,  via`HttpConfiguration#setOutputBufferSize`

That's the whole point of why we fixed #12963

Reply:
And if 12.1.x already has it, why not just make this change there?

Reply:
We are constrained in the changes we make  to 12.0 in a way that we are not in 12.1